### PR TITLE
use temporary files instead of hard-coded in TTS

### PR DESCRIPTION
**Pull request by norbusan**
When using flite or watson TTS, temporary files are created in
the CWD and `extra` subdirectory of CWD, but that might not
exist on desktop installations.

Use `tempfile` module to create temporary directories and clean them
up automatically when processing is done.

### CODE REVIEW

File: main/speech/TTS.py [hunk 1]
```diff
@@ -25,15 +26,17 @@ def speak_flite_tts(text):
...
-        ['flite', '-v', '-voice', 'file://' + flite_speech_file, '-f', filename, '-o', 'extras/output.wav'])   # nosec #pylint-disable type: ignore
-    subprocess.call(['play', 'extras/output.wav'])   # nosec #pylint-disable type: ignore
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        fd, filename = tempfile.mkstemp(text=True, dir=tmpdirname)
+        with open(fd, 'w') as f:
+            f.write(text)
+        # Call flite tts to reply the response by Susi
+        flite_speech_file = config['flite_speech_file_path']
+        logger.debug('flite -voice file://%s -f %s', flite_speech_file, filename)
+        fdout, wavOutput = tempfile.mkstemp(suffix='.wav', dir=tmpdirname)
```
Comment (hongquan) [inline main/speech/TTS.py]: Could you please try to follow Python coding style in naming variable? (prefer `lower_case_thing` to `someThing`).

Comment (norbusan) [reply main/speech/TTS.py]: Fixed in commit 443a45700fba7438983aa98cb09fc63426e9a64f

**Comment by mariobehling** [comment]
Please resolve conflicts.

**Comment by norbusan** [comment]
Merged development branch

**Pull request merged**